### PR TITLE
[6.2] refactor(tipfotheday): add dialog class

### PR DESCRIPTION
**Pull request by miurahr**
- Set default on close button
- set InputMap; left, right, escape

## Pull request type

- Bug fix -> [bug]

## Other information

TipOfTheDay dialog is disabled in 6.1 development cycle so the change does not any sub effect in 6.1 release.

### CODE REVIEW

**Comment by miurahr** [comment]
There are several checkstyle warnings that should be fixed before merge.

**Comment by miurahr** [comment]
Now wait #640 merged.

**Review by brandelune — APPROVED** [review]

**Comment by miurahr** [comment]
I'd like to rebased on master.

**Comment by Kazephil** [comment]
Is there a setting that's required for this feature?

I was going to take a look at it and comment, but I don't see a Tip of the Day dialog when I run the code for this branch.

I tried both the Run and InstallDist Gradle tasks.

**Comment by miurahr** [comment]
> Is there a setting that's required for this feature?
> 
> I was going to take a look at it and comment, but I don't see a Tip of the Day dialog when I run the code for this branch.
> 
> I tried both the Run and InstallDist Gradle tasks.

It is temporary dsiabled and will be enabled in 6.2 development cycle,  by a request from project lead @brandelune
https://sourceforge.net/p/omegat/mailman/omegat-development/thread/274E0297-94BC-47F0-8D98-03AD954EBF44%40traduction-libre.org/#msg37844136

You can force to enable it with editing `omegat.prefs` with entry `<tipoftheday_show_on_start>true</tipoftheday_show_on_start>`

or temporary removal of comment out to enable the menu item at
https://github.com/omegat-org/omegat/pull/592/files#diff-85f38ec4ab5212b891e023457d6f7854c0e66fe641771f57703974960f28d8b7L60-L61
```
                    // FIXME: disable temporary for 6.1 release
                    // initMenu();
```
and look at  `HELP> show tipoftheday`  menu.  @Kazephil

**Comment by miurahr** [comment]
This is a change which feature has not been enabled yet, so we can merge it without warring.

File: tipoftheday/src/main/java/org/omegat/gui/tipoftheday/TipOfTheDayController.java [hunk 1]
```diff
@@ -77,16 +80,16 @@ private void initUI() {
...
                 totdMenu.setToolTipText(UIManager.getDefaults().getString("TipOfTheDay.menuToolTipText"));
-                // force to show Tip of the Day dialog.
+                // show Tip of the Day dialog on startup.
                 totdMenu.addActionListener(actionEvent -> TipOfTheDayController.start(true));
                 MenuExtender.addMenuItem(MenuKey.HELP, totdMenu);
             }
 
             @Override
             public void onApplicationShutdown() {
+                MenuExtender.removeMenuItems(MenuKey.HELP, Collections.singletonList(totdMenu));
```
Comment (miurahr) [inline tipoftheday/src/main/java/org/omegat/gui/tipoftheday/TipOfTheDayController.java:92]: When disabled menu item, it will cause some error.

**Pull request merged**